lib/core/public/cleanup-plugins.js

```diff
@@ -23,14 +23,18 @@ function cleanupPlugins(resolve, reject) {
     });
   });
 
-  axe.utils.toArray(document.querySelectorAll('frame, iframe')).forEach(function (frame) {
+  var flattenedTree = axe.utils.getFlattenedTree(document.body)
+
+  axe.utils.querySelectorAll(flattenedTree, 'iframe, frame').forEach(function (node) {
     q.defer(function (res, rej) {
-      return axe.utils.sendCommandToFrame(frame, {
+      return axe.utils.sendCommandToFrame(node.actualNode, {
         command: 'cleanup-plugin'
       }, res, rej);
     });
   });
 
+  flattenedTree = undefined

```

Comment:
Will do. I'm curious, though, why we need to manually free up the memory here: https://github.com/dequelabs/axe-core/blob/develop/lib/core/base/audit.js#L182